.github/workflows/python-app.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python application
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
```

Comment:
I would add these to `dev-requirements.txt` and then we can just pip install that just like `requirements.txt`

Reply:
Okay so I should create a separate PR adding flake8 to dev-requirements?

Reply:
I would just do it in this PR

Reply:
Okay I added flake8- there's a conflict now though with main branch. Did you commit these suggestions? I don't seem to have the usual option

Reply:
Yeah this branch is apparently six commits behind main...

Reply:
I didn't commit any suggestions no, you could use a merge to bring this branch up to date with master (see https://bioconductor.org/developers/how-to/git/pull-upstream-changes/)